File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
          int k = 0;
        int same = nums[0];
        for(int i=1;i<nums.length;i++){
            if(nums[i]!=same){
                same=nums[i];
                nums[i-k]=nums[i];
            }
            else{
                k++;
            }
        }
        return nums.length-k;
    
        
    }
}

File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) ==0 : return 0
        slow = 0
        for fast in range(1,len(nums)):
            if nums[slow] != nums[fast]:
                slow += 1
                nums[slow] = nums[fast]
        return slow+1

        
        
        
        
        
        #brute force
#         dup = sorted(set(nums))
#         i = 0
#         for x in dup:
#             nums[i] = x
#             i+=1
       
#         while(i<len(nums)):
#             nums[i] = None
#             i+=1
            
     
#         return len(dup)
            
        